graphql-cli/src/main/java/io/ballerina/graphql/cmd/GraphqlCmd.java

```diff
@@ -208,21 +228,32 @@ private void validateInputFlags() throws CmdException {
         if (useRecordsForObjectsFlag && !(inputPath.endsWith(GRAPHQL_EXTENSION))) {
             throw new CmdException(String.format(Constants.MESSAGE_FOR_USE_RECORDS_FOR_OBJECTS_FLAG_MISUSE, mode));
         }
+        
+        if (refreshFlag && !(inputPath.endsWith(GRAPHQL_EXTENSION)) && 
+            !(mode == null || MODE_SERVICE.equals(mode))) {
+            throw new CmdException("Refresh flag can only be used with GraphQL schema files and service mode");
+        }
     }
 
     private boolean validInputFileExtension(String filePath) {
-        return filePath.endsWith(YAML_EXTENSION) || filePath.endsWith(YML_EXTENSION) ||
-                filePath.endsWith(BAL_EXTENSION) || filePath.endsWith(GRAPHQL_EXTENSION);
+        if (filePath == null || filePath.trim().isEmpty()) {
+            return false;
+        }
+        
+        String trimmedPath = filePath.trim();
+        return trimmedPath.endsWith(YAML_EXTENSION) || trimmedPath.endsWith(YML_EXTENSION) ||
+                trimmedPath.endsWith(BAL_EXTENSION) || trimmedPath.endsWith(GRAPHQL_EXTENSION);
     }
 
     private boolean isModeCompatible() throws CmdException {
-        if (mode != null) {
-            if (MODE_CLIENT.equals(mode)) {
-                return inputPath.endsWith(Constants.YAML_EXTENSION) || inputPath.endsWith(Constants.YML_EXTENSION);
-            } else if (MODE_SCHEMA.equals(mode)) {
-                return inputPath.endsWith(Constants.BAL_EXTENSION);
-            } else if (MODE_SERVICE.equals(mode)) {
-                return inputPath.endsWith(Constants.GRAPHQL_EXTENSION);
+        if (mode != null && !mode.trim().isEmpty()) {
```

Comment:
Redundant checks. `trim` is not needed here. Empty strings will be handled by the PicoCLI.